src/main/java/org/primefaces/component/linkbutton/LinkButton.java

```diff
@@ -45,30 +49,36 @@
         return ComponentUtils.getUIParams(this);
     }
 
-    public String resolveStyleClass() {
-        String icon = getIcon();
-        Object value = getValue();
-        String styleClass = "";
+    public String resolveStyleClass(FacesContext context) {
+        boolean hasIcon = LangUtils.isNotBlank(getIcon());
+        boolean hasValue = LangUtils.isNotBlank((String) getValue()) || hasDisplayedChildren();
+        boolean isTextAndIcon = hasValue && hasIcon;
 
-        if (value != null && LangUtils.isValueBlank(icon)) {
-            styleClass = HTML.BUTTON_TEXT_ONLY_BUTTON_CLASS;
-        }
-        else if (value != null && !LangUtils.isValueBlank(icon)) {
-            styleClass = getIconPos().equals("left") ? HTML.BUTTON_TEXT_ICON_LEFT_BUTTON_CLASS : HTML.BUTTON_TEXT_ICON_RIGHT_BUTTON_CLASS;
-        }
-        else if (value == null && !LangUtils.isValueBlank(icon)) {
-            styleClass = HTML.BUTTON_ICON_ONLY_BUTTON_CLASS;
-        }
+        String styleClass = PrimeRequestContext.getCurrentInstance(context).getStyleClassBuilder()
+                    .add("ui-linkbutton")
+                    .add(getStyleClass())
+                    .add(hasValue && !hasIcon, HTML.BUTTON_TEXT_ONLY_BUTTON_CLASS)
+                    .add(!hasValue && hasIcon, HTML.BUTTON_ICON_ONLY_BUTTON_CLASS)
+                    .add(isTextAndIcon && getIconPos().equals("left"), HTML.BUTTON_TEXT_ICON_LEFT_BUTTON_CLASS, HTML.BUTTON_TEXT_ICON_RIGHT_BUTTON_CLASS)
+                    .add(isDisabled(), "ui-state-disabled")
+                    .build();
 
-        if (isDisabled()) {
-            styleClass = styleClass + " ui-state-disabled";
-        }
+        return styleClass;
+    }
 
-        String userStyleClass = getStyleClass();
-        if (userStyleClass != null) {
-            styleClass = styleClass + " " + userStyleClass;
+    public boolean hasDisplayedChildren() {
+        if (getChildCount() == 0) {
+            return false;
         }
-
-        return "ui-linkbutton " + styleClass;
+        else {
+            for (int i = 0; i < getChildCount(); i++) {
+                UIComponent child = getChildren().get(i);
+                if (child instanceof UIParameter || !child.isRendered()) {
```

Comment:
what about returning true if child not UIParameter? I just feel is more readable and straightforward that way